tvtk/wrapper_gen.py

```diff
@@ -8,24 +8,185 @@
+        cols = len(value)
+        if kwargs:
+            kwargs += ', '
+
+        kwargs += ('shape={shape}, dtype={dtype}, '
+                   'value={value!r}, cols={cols}').format(
+                       shape=shape, dtype=dtype,
+                       value=value, cols=min(3, len(value)))
+
+        return 'traits.Array', '', kwargs
+
+    elif type_ is bool:
+        return 'traits.Bool', '', kwargs
+
+    else:
+        raise TypeError("Could not understand type: {}".format(type_))
+
+
+def patch_default(vtk_get_meth, vtk_set_meth, default):
+    """Patch the initial default value for an attribute of
+    a VTK class that does not initialise it properly.
+
+    Parameters
+    ----------
+    vtk_get_meth : Method for getting the position attribute
+
+    vtk_set_meth : Method for setting the position attribute
+
+    default : initial default value
+
+    Returns
+    -------
+    default
+       If patching fails, the initial default is returned
+
+    Examples
+    --------
+    >>> import vtk
+    >>> vtk.vtkVersion.GetVTKVersion()
+    '6.3.0'
+    >>> obj = vtk.vtkXOpenGLRenderWindow()
+    >>> obj.GetPosition()
+    '_000000000351c458_p_void'
+
+    >>> patch_default(vtk.vtkXOpenGLRenderWindow.GetPosition,
+                      vtk.vtkXOpenGLRenderWindow.SetPosition,
+                      '_000000000351c458_p_void')
+    (0, 0)
+    """
+    # We will attempt to guess the default by looking into the
+    # arguments of the Set method
+    # SetPosition(int, int) has a signature of ("int", "int")
+    # SetPosition(int position[2]) has a signature of (["int", "int"],)
+    # Some method even has a signature of (["int", "int"], "vtkInformation")
+    arg_formats = []
+
+    # Collect the signatures of the get method
+    # We only use the arguments
+    all_sigs = vtk_parser.VTKMethodParser.get_method_signature(vtk_get_meth)
+
+    # Collect the signatures of the set method
+    all_sigs.extend(
+        vtk_parser.VTKMethodParser.get_method_signature(vtk_set_meth))
+
+    for sig in all_sigs:
+        if sig[1] is None:
+            continue
+
+        if len(sig[1]) == 1:
+            # This unpacks tuple of something e.g. (('int', 'int', 'int'))
+            arg_formats.append(tuple(chain.from_iterable(sig[1])))
+
+        arg_formats.append(tuple(sig[1]))
+
+    default_mappings = {
+        'int' : 0,
+        'float': 0.0,
+        'string': ''
+        }
+
+    for arg_format in arg_formats:
+        try:
+            all_same_type = len(set(arg_format)) == 1
+        except TypeError:  # Unhashable
+            continue
+
+        if all_same_type and arg_format[0] in default_mappings:
+            # All types in `arg_format` are the same and they are
+            # in the mapping (e.g. arg_format = ('int', 'int')
+            default = default_mappings[arg_format[0]]
+
+            if len(arg_format) > 1:
+                return (default,)*len(arg_format)
```

Comment:
For those cases, the vtk set method and get method are written as is.  This is exactly what `test_spider_plot_actor` in `tvtk/test_tvtk.py` is testing.